app/ui/src/app/integrations/edit-page/action-select/action-select.component.ts

```diff
@@ -101,16 +102,48 @@ export class IntegrationsSelectActionComponent extends FlowPage
     }
   }
 
+  // filters out actions that do not use the 'From' pattern
+  connectorStartActionsFilter(connector: Connector): Connector {
+    const filteredConnectorActions = connector.actions.filter(action => action.pattern === 'From');
+    connector.actions = filteredConnectorActions;
```

Comment:
Would this overwrite the `connector.actions` property so when the connector is used for the second time at a different place in the integration then those actions might be missing?

I might be talking nonsense, I'm TypeScript/Angular newb.

Reply:
It's a good question, I'm not 100% sure. What's a good way to test this?

Reply:
I don't think it would, because if it did we'd be getting a twice filtered list in the "Finish" step, but we don't. I think that filter is only applied within the context of a single view. Would be good to have someone verify this, or let me know how I can just to be sure.

Reply:
It would actually overwrite it, I'm afraid. Remember that JavaScript operates by reference, and JavaScript objects are mutable (this is by design on JavaScript: TypeScript and Angular are unrelated to this design principle), so referencing that object to a variable and changing its property will change that property elsewhere in other reference application-wide. 

Correct me if I'm wrong, but apparently what you pursue here is to fetch all connectors that are undefined (including null connectors would be useful too) and then, in the context of `OnInit`, assign the actions from the `connector`object whose `pattern` property is set to `From` to the `actions` property in the component, right?

After quickly assessing the code and borrowing some help from the RxJS operands,  I'd suggest to apply this other approach straight on LOC 124 (hence removing the `connectorStartActionsFilter()` method, which should have been flagged as `private` anyways):

```typescript
if (this.currentStep === this.currentFlow.getFirstPosition()) {
  this.actions = this.connector
    .filter(connector => connector)
    .switchMap(connector => connector.actions.filter(action => action.pattern === 'From'));
}
```

Reply:
AS I mentioned elsewhere, you'll probably want to import the `switchMap` operand into the component in order to be able to use it. Try to implicitly use it first, and should the compiler complains, then add this line at the top, right after the angular imports:

`import 'rxjs/add/operator/switchMap';`

Ideally we should have been adding RxJS operands from a centralized vendor file generated as a bundle by either the Angular CLI or Webpack since the project's kickstart, so this should not be necessary.